### PR TITLE
Category Swap: Fix special cases and rulify

### PR DESCRIPTION
**Pull request by WeWuzNidokangz**
The special casing is needed to make certain moves follow the OM rule correctly. Rulification is just to support mashups better and to future-proof mechanics support in case the format is removed after LCotM.

### CODE REVIEW

**Comment by KrisXV** [comment]
These special cases for photon geyser/clones are not mentioned anywhere in the op of the thread and they were not mentioned while the meta was in the submissions stage. I also didn’t make this a rule on purpose.

**Comment by KrisXV** [comment]
> These special cases for photon geyser/clones are not mentioned anywhere in the op of the thread and they were not mentioned while the meta was in the submissions stage. I also didn’t make this a rule on purpose.

Btw with this I’m saying make sure to clarify how they work in the op before I merge this

File: data/rulesets.ts [hunk 1]
```diff
@@ -1979,4 +1979,96 @@ export const Rulesets: {[k: string]: FormatData} = {
 			}
 		},
 	},
+	categoryswaprule: {
```
Comment (KrisXV) [inline data/rulesets.ts:1982]:
```suggestion
	categoryswapmod: {
```

File: data/rulesets.ts [hunk 2]
```diff
@@ -1979,4 +1979,96 @@ export const Rulesets: {[k: string]: FormatData} = {
...
+		effectType: 'Rule',
+		name: 'Category Swap Rule',
```
Comment (KrisXV) [inline data/rulesets.ts:1984]:
```suggestion
		name: 'Category Swap Mod',
```

File: data/rulesets.ts [hunk 3]
```diff
@@ -1979,4 +1979,96 @@ export const Rulesets: {[k: string]: FormatData} = {
...
+		desc: `All physical moves become special, and all special moves become physical.`,
+		onBegin() {
+			this.add('rule', 'Category Swap Rule: All physical moves become special, and all special moves become physical');
```
Comment (KrisXV) [inline data/rulesets.ts:1987]:
```suggestion
			this.add('rule', 'Category Swap Mod: All physical moves become special, and vice versa');
```

**Comment by WeWuzNidokangz** [comment]
> These special cases for photon geyser/clones are not mentioned anywhere in the op of the thread and they were not mentioned while the meta was in the submissions stage. I also didn’t make this a rule on purpose

Okay, I'll ask for that to be checked. It was reported as a bug to me before without special-casing, but I don't remember if it was somebody on council who reported/confirmed it.

https://www.smogon.com/forums/threads/category-swap.3702709/page-3#post-9293421

Is there any benefit to _not_ making this a rule? It seems likely that it will become necessary to maintain the functionality at some point, while in the meantime having the mechanics entangled with the format just makes things like Category Swap AG, Ubers, etc unnecessarily convoluted to express and maintain.

File: data/rulesets.ts [hunk 4]
```diff
@@ -1979,4 +1979,96 @@ export const Rulesets: {[k: string]: FormatData} = {
...
+							effectType: 'Move',
+							isFutureMove: true,
+							type: 'Steel',
+						},
+					});
+					this.add('-start', source, 'Doom Desire');
+					return this.NOT_FAIL;
+				};
+			}
+			break;
```
Comment (KrisXV) [inline data/rulesets.ts:2022]: This needs to be in the brackets

**Comment by KrisXV** [comment]
I discuss making these rules with the om mod team at the end of each month, so we haven’t had a chance to discuss it yet

**Pull request merged**